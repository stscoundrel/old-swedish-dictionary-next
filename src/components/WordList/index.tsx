import WordLink from 'components/WordLink'
import { DictionaryEntry, DictionaryEntryDTO } from 'lib/models/dictionary'
import styles from './WordList.module.scss'

interface WordListProps{
  words: DictionaryEntry[] | DictionaryEntryDTO[]
}

export default function WordList({ words }: WordListProps) {
  return (
    <ul className={styles.list}>
      { words.map((word) => (
        <li key={word.slug}>
          <WordLink data={word} useLowerCase={true} />
        </li>
      )) }
    </ul>
  )
}
