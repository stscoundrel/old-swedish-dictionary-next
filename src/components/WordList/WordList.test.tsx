import ReactDOM from 'react-dom/client'
import renderer from 'react-test-renderer'
import WordList from './index'

const words = [
  {
    headword: 'afköra',
    partOfSpeech: ['vb'],
    grammaticalAspect: 'v.',
    definitions: [
      ' , vräka. hafwi wald han afköra  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.',
    ],
    alternativeForms: ['-bortdrifva'],
    slug: 'afkora',
  },
  {
    headword: 'afköra 2',
    partOfSpeech: ['vb'],
    grammaticalAspect: 'v.',
    definitions: [
      ' , vräka. hafwi wald han afköra  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.',
    ],
    alternativeForms: ['-bortdrifva'],
    slug: 'afkora-2',
  },
  {
    headword: 'afköra 3',
    partOfSpeech: ['vb'],
    grammaticalAspect: 'v.',
    definitions: [
      ' , vräka. hafwi wald han afköra  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.',
    ],
    alternativeForms: ['-bortdrifva'],
    slug: 'afkora-3',
  },
  {
    headword: 'afköra 4',
    partOfSpeech: ['vb'],
    grammaticalAspect: 'v.',
    definitions: [
      ' , vräka. hafwi wald han afköra  GS 43 (1416?). af ty ahren (för areno) han landboen afkörer ib.',
    ],
    alternativeForms: ['-bortdrifva'],
    slug: 'afkora-4',
  },
]

describe('WordList component', () => {
  test('Does not crash', () => {
    const div = document.createElement('div')
    const root = ReactDOM.createRoot(div)
    root.render(<WordList words={words} />)
  })

  test('Matches snapshot', () => {
    const tree = renderer.create(<WordList words={words} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Renders correct amount of words', () => {
    const tree = renderer.create(<WordList words={words} />)
    const { root } = tree

    expect(root.findAllByType('ul').length).toEqual(1)
    expect(root.findAllByType('li').length).toEqual(4)
  })
})
