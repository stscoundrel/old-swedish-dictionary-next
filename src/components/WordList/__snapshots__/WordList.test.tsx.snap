// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WordList component Matches snapshot 1`] = `
<ul
  className="list"
>
  <li>
    <a
      href="/word/afkora"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      afköra
    </a>
  </li>
  <li>
    <a
      href="/word/afkora-2"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      afköra 2
    </a>
  </li>
  <li>
    <a
      href="/word/afkora-3"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      afköra 3
    </a>
  </li>
  <li>
    <a
      href="/word/afkora-4"
      onClick={[Function]}
      onMouseEnter={[Function]}
      onTouchStart={[Function]}
    >
      afköra 4
    </a>
  </li>
</ul>
`;
